sensor: ms5837: fix compensate parameters for 30BA variant

EthanAussie opened the pull request:
The previous parameters seems wrong if we refer to: 
https://www.te.com/usa-en/product-CAT-BLPS0017.html

![compensation from data sheet](https://github.com/zephyrproject-rtos/zephyr/assets/50738558/c34e7ff4-768e-4d5b-b193-7b17017d4677)

kartben commented:
please amend your commit so that your author first/last name matches the signed-off by entry. Thanks for the PR!

EthanAussie commented:
Not sure what is the status of this pull request.
I just rebased to the latest Zephyr.
Hope this helps.

kartben reviewed (dismissed):
@EthanAussie sorry that this felt through the cracks -- lgtm

@teburd @MaureenHelm @ubieda maybe you can help review too - thx!

ubieda commented on drivers/sensor/meas/ms5837/ms5837.c:
```diff
@@ -75,17 +75,17 @@ static void ms5837_compensate_30(const struct device *dev,
-		Ti = (3ll * dT * dT) / (1ll << 23);
-		OFFi = (3ll * temp_sq) / 1ll;
+		Ti = (3ll * dT * dT) / (1ll << 33);
+		OFFi = (3ll * temp_sq) / 1ll << 1;
```
We need to add parenthesis here, otherwise division will take precedence to the left-shift operation.
```suggestion
		OFFi = (3ll * temp_sq) / (1ll << 1);
```

EthanAussie replied:
Thanks, that is a great catch.
I am going to change this and submit again.
Question is: normally, what is the best practices to resubmit it?
The latest zephyr main moved MS5837 to new vendor folder.
Should I rebase on the latest zephyr main branch?

EthanAussie replied:
First time to do the pull request and change, etc.
I changed the original branch according to ubieda's suggestion. 
It is great catch. Thanks.

ubieda reviewed (changes requested):
Thanks for your contribution!
AFAIK - Just a small change-req, otherwise looks good!

EthanAussie committed:
sensor: ms5837: fix compensate parameters for 30BA variant

The previous parameters seems wrong if we refer to:
https://www.te.com/usa-en/product-CAT-BLPS0017.html

Signed-off-by: Ethan Lu <ethan@cpdesign.com.au>
